zio-http-example/src/main/scala/example/HtmlTemplating.scala

```diff
@@ -46,5 +46,5 @@ object HtmlTemplating extends ZIOAppDefault {
     }
   }
 
-  def run = Server.start(8090, app)
+  def run = Server.serve(app).provide(Server.default)
```

Comment:
Hmm, we have a problem: these apps will exit immediately.

You should probably add `Server.serveForever(app)` defined as `Server.serve(app) *> ZIO.never`.

Or maybe that should go to `serve` and we could change the existing `server` into `add` or `install`. So if you just want to serve 1 app forever, you do `Server.serve`, but if you want to install a server (e.g. from another layer which requires to expose some endpoints), then you do `Server#install`.

Reply:
No, it will not exist right away because the `serve` implementation is:
`  def serve[R](httpApp: HttpApp[R, Throwable]): URIO[R with Server, Unit] =
    ZIO.serviceWithZIO[Server](_.serve(httpApp)) *> ZIO.never`

Reply:
Hmmm, we need two different methods because both are useful

Reply:
In fact, let's have `install` be the only method on Server, then `serve` can be a convenience method on `Server` companion object that installs and waits forever (via never)

Reply:
done